bundle/src/main/java/com/adobe/acs/commons/wcm/properties/shared/impl/SharedComponentPropertiesBindingsValuesProvider.java

```diff
@@ -57,107 +55,81 @@
 public class SharedComponentPropertiesBindingsValuesProvider implements BindingsValuesProvider {
     private static final Logger log = LoggerFactory.getLogger(SharedComponentPropertiesBindingsValuesProvider.class);
 
-    @Reference
-    private PageRootProvider pageRootProvider;
-
-    @Reference
+    /**
+     * Bind if available, check for null when reading.
+     */
+    @Reference(policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.OPTIONAL_UNARY)
     private SharedComponentProperties sharedComponentProperties;
 
     @Override
-    public void addBindings(Bindings bindings) {
-        Resource resource = (Resource) bindings.get("resource");
-        if (resource != null) {
-            if (pageRootProvider != null) {
-                setSharedProperties(bindings, resource);
+    public void addBindings(final Bindings bindings) {
+        final SlingHttpServletRequest request = (SlingHttpServletRequest) bindings.get(SlingBindings.REQUEST);
+        final Resource resource = (Resource) bindings.get(SlingBindings.RESOURCE);
+        if (request != null && resource != null) {
```

Comment:
Previous code only checked for `resource != null` but now we're also requiring `request` to be non-null.  Does this add any risks?

Reply:
No. A BindingsValuesProvider is only executed in the context of a Sling request, and should have access to all property names defined by the SlingBindings interface. But we still want to check for null, in case, you know, whatever.